include/RAJA/util/View.hpp

```diff
@@ -296,6 +297,48 @@ RAJA_INLINE AtomicViewWrapper<ViewType, AtomicPolicy> make_atomic_view(
   return RAJA::AtomicViewWrapper<ViewType, AtomicPolicy>(view);
 }
 
+struct layout_left{};
+struct layout_right{};
+
+template<typename LAYOUT>
+struct Reshape;
+
+template<>
+struct Reshape<layout_right>
+{
+  template<typename T, typename...Ts>
+  static auto get(T *ptr, Ts... s)
+  {
+    constexpr int N = sizeof...(Ts);
+    using view_t = RAJA::View<T, RAJA::Layout<N, RAJA::Index_type, N-1>>;
+
+    return view_t(ptr, s...);
+  }
+};
+
+template<std::size_t... Is>
+constexpr std::array<RAJA::idx_t, sizeof...(Is)> make_reverse_array(std::index_sequence<Is...>) {
+  return std::array<RAJA::idx_t, sizeof...(Is)>{sizeof...(Is) - 1U - Is ...};
+}
+
+template<>
+struct Reshape<layout_left>
+{
+  template<typename T, typename...Ts>
+  static auto get(T *ptr, Ts... s)
+  {
+  constexpr int N = sizeof...(Ts);
+
+  std::array<RAJA::idx_t, N> extent{s...};
+
+  constexpr auto reverse_array = make_reverse_array(std::make_index_sequence<N>{});
```

Comment:
@MrBurmark this is the revised implementation

Reply:
Nice, this is what I was thinking.